crates/astria-sequencer-client/src/client.rs

```diff
@@ -0,0 +1,458 @@
+
+/// Tendermint HTTP client which is used to interact with the Sequencer node.
+pub struct Client {
+    client: HttpClient,
+}
+
+impl Client {
+    /// Creates a new Tendermint client with the given base URL.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the client could not be created.
+    pub fn new(base_url: &str) -> eyre::Result<Self> {
+        Ok(Client {
+            client: HttpClient::new(base_url).wrap_err("failed to initialize tendermint client")?,
+        })
+    }
+
+    /// Returns the latest block.
+    ///
+    /// # Errors
+    ///
+    /// - If calling the tendermint RPC endpoint fails.
+    pub async fn get_latest_block(&self) -> eyre::Result<BlockResponse> {
+        let block = self
+            .client
+            .latest_block()
+            .await
+            .wrap_err("failed to call latest_block")?;
+        Ok(block)
+    }
+
+    /// Returns the block at the given height.
+    ///
+    /// # Errors
+    ///
+    /// - If calling the tendermint RPC endpoint fails.
+    pub async fn get_block(&self, height: Height) -> eyre::Result<BlockResponse> {
+        let block = self
+            .client
+            .block(height)
+            .await
+            .wrap_err("failed to call block")?;
+        Ok(block)
+    }
+
+    /// Returns the balance of the given account at the given height.
+    /// If no height is given, the latest height is used.
+    ///
+    /// # Errors
+    ///
+    /// - If calling the tendermint RPC endpoint fails.
+    /// - If the response from the server is invalid.
+    pub async fn get_balance(
+        &self,
+        address: &Address,
+        height: Option<Height>,
+    ) -> eyre::Result<Balance> {
+        let response = self
+            .client
+            .abci_query(
+                Some(format!("accounts/balance/{}", &address.to_string())),
+                vec![],
+                height,
+                false,
+            )
+            .await
+            .wrap_err("failed to call abci_query")?;
+
+        let balance = QueryResponse::try_from_slice(&response.value)
+            .wrap_err("failed to deserialize balance bytes")?;
+
+        if let QueryResponse::BalanceResponse(balance) = balance {
+            Ok(balance)
+        } else {
+            bail!("received invalid response from server: {:?}", &response);
+        }
+    }
+
+    /// Returns the nonce of the given account at the given height.
+    /// If no height is given, the latest height is used.
+    ///
+    /// # Errors
+    ///
+    /// - If calling the tendermint RPC endpoint fails.
+    /// - If the response from the server is invalid.
+    pub async fn get_nonce(
+        &self,
+        address: &Address,
+        height: Option<Height>,
+    ) -> eyre::Result<Nonce> {
+        let response = self
+            .client
+            .abci_query(
+                Some(format!("accounts/nonce/{}", &address.to_string())),
```

Comment:
Isn't adress implementing Display? If is you should throw it directly into the format!